src/index.js

```diff
@@ -1,43 +1,110 @@
-  },
-  {
-    description: 'Finish my To do App',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Learn new JS FrameWorks',
-    completed: false,
-    index: 2,
-  },
-
-];
-const loadTask = () => {
-  const sortedList = LIST.sort((item1, item2) => item1.index - item2.index);
-
-  for (let i = 0; i < sortedList.length; i += 1) {
-    listContainer.innerHTML += `
-    
-    <div class="task-container">
-      <form class="task-form">
-      
-        <input type="checkbox" class="checkbox" <p class="task-description">&nbsp&nbsp${sortedList[i].description} 
-        <i class="fas fa-ellipsis-v pointer edit-task" aria-hidden="true"></i>
-        </p>
-        
-      </form>
-  </div>
-  
-    `;
+let todoTask = [];
+const taskWrapper = document.querySelector('.todo-activities');
+const newTask = document.getElementById('task-description');
+const newTaskBtn = document.getElementById('new-task');
+const reset = document.getElementById('refresh');
+
+const clearallTask = () => {
+  todoTask = [];
+};
+
+const addListToLocalStorage = () => {
+  localStorage.setItem('myTodoTasks', JSON.stringify(todoTask));
+};
+const getListFromLocalStorage = () => {
+  if (localStorage.getItem('myTodoTasks')) {
+    todoTask = JSON.parse(localStorage.getItem('myTodoTasks'));
+  }
+  return todoTask;
+};
+
+const refreshItems = (todoTask) => {
+  for (let i = 0; i < todoTask.length; i += 1) {
+    const indexes = i + 1;
+    todoTask[i].index = indexes;
+  }
+};
+// remove/Delete task
+const removeTask = (index) => {
+  const myLocalStorage = getListFromLocalStorage();
+  myLocalStorage.splice(index - 1, 1);
+};
+// Edit Task
+const editTask = (taskdescription, index) => {
+  for (let j = 0; j < todoTask.length; j += 1) {
+    if (todoTask[j] === index) {
+      todoTask[j] += '*';
+    }
   }
+  todoTask[index - 1].description = taskdescription;
+  addListToLocalStorage();
 };
+// Read Items
+const createTask = () => {
+  taskWrapper.innerHTML = '';
+  const mylocal = getListFromLocalStorage();
+  mylocal.forEach((task) => {
+    const li = document.createElement('li');
+    const checkbox = document.createElement('input', 'fas', 'fa-trash-can');
+    checkbox.setAttribute('type', 'checkbox');
+    if (task.checked === true) {
+      checkbox.setAttribute('checked', 'checked');
+    }
+
+    const taskDesc = document.createElement('input');
+    taskDesc.classList.add('todotask');
+    taskDesc.value = task.description;
+
+    const deleteTask = document.createElement('i');
+    taskDesc.addEventListener('change', (e) => {
+      e.preventDefault();
+      editTask(e.target.value, task.index);
+      taskDesc.blur();
+    });
+    deleteTask.classList.add('fas', 'fa-trash-can');
+    deleteTask.addEventListener('click', () => {
+      removeTask(task.index[0]);
+      refreshItems(mylocal);
+      addListToLocalStorage();
+      createTask();
+    });
+
+    li.append(checkbox, taskDesc, deleteTask);
+    taskWrapper.appendChild(li);
```

Comment:
- [ ] When deleting a task you should update the index of the remaining tasks so that will be in order as required 
<img width="1081" alt="Screenshot 2023-01-25 at 15 35 23" src="https://user-images.githubusercontent.com/61790587/214591648-a94c0ecf-fdfe-4025-815d-5cd1dd19428e.png">
<img width="887" alt="Screenshot 2023-01-25 at 15 34 05" src="https://user-images.githubusercontent.com/61790587/214591706-ad981b3f-55cd-4d29-81c2-2a4df0581786.png">